### PR TITLE
API types and basic controller/actuator scaffolding

### PR DESCRIPTION
**Pull request by detiber**
Adds the initial basic API types and minimal scaffolding for the controller/actuators.

### CODE REVIEW

**Review by displague — COMMENTED** [review]
I'm mainly interested in the starting point that this PR provides for the Equinix Metal provider types that will be used by https://github.com/openshift/installer/pull/3914.

https://github.com/openshift/cluster-api-provider-packet/pull/2/files#diff-842042708803b07649ebe3065dcd5adcf03fdbe36b289817840a4c27aab2c722R15-R27

This is a sufficient starting point upon which we can build.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	k8s.io/apimachinery v0.19.0
+	k8s.io/client-go v0.19.0
+	k8s.io/klog/v2 v2.3.0
+	sigs.k8s.io/controller-runtime v0.6.2
+	sigs.k8s.io/controller-tools v0.3.0
+	sigs.k8s.io/yaml v1.2.0
+)
+
+replace (
+	sigs.k8s.io/cluster-api-provider-aws => github.com/openshift/cluster-api-provider-aws v0.2.1-0.20200929152424-eab2e087f366
```
Comment (displague) [inline go.mod:22]: I'm curious, are these modules actually needed in this provider?  Many of the vendored files are related to these clouds.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-openshift-4.6 AS builder
+WORKDIR /go/src/github.com/openshift/cluster-api-provider-packet
```
Comment (displague) [inline Dockerfile:2]: Who can help us rename this project to `cluster-api-provider-equinix-metal`? (or unhyphenated `equinixmetal`, I don't know if there is a preference for technical or historical reasons)

File: go.mod (hunk 1 above)
Comment (detiber) [reply go.mod:22]: Following up on this, it looks like they are pulled in from the machine-api-operator dependency and causes failures if not specified.

**Comment by detiber** [comment]
Closing this in favor of #1

**Pull request closed without merging**